built-ins/authoring-built-ins.md

```diff
@@ -0,0 +1,132 @@
+  * *"Prod"* - Azure **public production cloud** (eg: https://portal.azure.com) used by most Azure customers.
+  * *"MC"* - Chinese sovereign cloud (eg: https://portal.azure.cn)
+  * *"FF"* - US Government cloud. Non-Microsoft employees can disregard this value.
+
+Additionally the following **optional** properties are available:
+
+* **supportsAirGapped**: A boolean value that indicates whether this built-in should be returned in airgapped cloud environments. Defaults to false. Expected to be set by Microsoft employees only.
+
+### Step 5: Submit a PR to the template specs repo
+
+> **IMPORTANT:** Authoring and contribution of Built-in Template Specs is restricted to **authorized collaborators only** such as internal Microsoft teams and partners. PRs from unauthorized individuals will be rejected with no exceptions.
+ 
+Now that you've created all the necessary changes locally in your repo/branch, you should commit your changes and initiate a pull request to the main repo.
+
+* For forked repos, you can see guidance for creating a PR here: [Creating a pull request from a fork - GitHub Docs](https://docs.github.com/en/github/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request-from-a-fork)
+* For branches within the template spec repo you can see guidance for creating a PR here: [Creating a pull request - GitHub Docs](https://docs.github.com/en/github/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request)
+
+Be sure to fill out the PR template that is shown with the relevant details. After your PR is received someone from Microsoft will review it and either provide feedback requesting changes, or it will be approved and rollout can be expected within 30 days. 
+
+## How do I update an existing built-in?
+
+All updates to existing built-ins should come as new versions, with exceptions for the following situations:
+* Simple text updates (no modifications of functionality)
+* Critical fixes for security purposes 
+
+Unless your update meets one of the exception scenarios above, you should submit a new version using the same steps outlined for a new built-in starting from: [Step 1: Validate your template/template spec functionality](#step-1:-validate-your-template/template-spec-functionality) . As part of the update process, be sure to bump your version in a way that will make sense to users (for example if your previous version name was "v1", your new version may be "v2").
+
+## I want to remove an existing built-in, what should I do?
+
+Built-ins should never be removed (doing so will break customers referencing the built-in during their deployments). We are currently working on implementing deprecation logic for this purpose. If you need a built-in removed for a **critical security issue** please contact templatespecs@microsoft.com with details.
```

Comment:
With the Bicep public module registry, we're following how other package managers (NuGet, NPM, etc.) do it and will only remove for legal reasons. Thoughts?

Reply:
I think we'll effectively be in a similar situation, but due to the internal partner restriction at the moment I don't think the legal aspect should be called out here.

"Critical security issue" here would be a highly unlikely (but not impossible) case where if we didn't pull a template spec, it could result in more deployments that result in compromised security (improperly configured security settings on a resource, etc).

Reply:
But should we remove a template spec and break users for security issues when all popular package managers don't do that at all?

Reply:
As we will start off by fully owning the ecosystem (internal contributions only, the Portal being the only UI, etc.) I think force removals have a higher benefit than leaving compromising built-ins as deployable. I'm expecting we won't see any critical security issues with built-ins (we never saw any with our built-in blueprints) but I want to leave the door open for damage control if we do, and IMO the best damage control is to block deployments of the compromised version (by removing it).

Reply:
I would leave it open to deal with this on a case-by-case basis. Removal seems like a method of last resort, but we could still use it as a nuclear option.